File: anisearchEx/loadResult.py
from anisearchEx.extractedSeason import ExtractedSeason
from anisearchEx.searchEntry import SearchEntry


class LoadResult(SearchEntry):
    def __init__(self,link) -> None:
        self.link = link
        self.name = ""
        self.image = ""
        self.desc = ""
        self.extractedSeasons = list()

    
    def loadImage(self, imageLink):
        self.image = self._downloadImage(imageLink)

File: src/anisearchEx/loadResult.ts
import { ExtractedSeason } from "./extractedSeason";
import { SearchEntry } from "./searchEntry";

export class LoadResult extends SearchEntry {
    desc:string;
    extractedSeasons: ExtractedSeason[];

    constructor(link:string) {
        super("","",link);
        this.desc = "";
        this.extractedSeasons = [];
    }

    async loadImageByLink(imageLink:string) {
        this.imageLink = imageLink;
        await this.loadImage();
    }
}